components/HackathonModal.tsx

```diff
@@ -0,0 +1,142 @@
+import React, { useState, useEffect } from 'react';
+import Modal from 'react-modal';
+import Image from '../components/Image';
+import Link from 'next/link';
+import styles from '../styles/components/HackathonModal.module.scss';
+import { website_config } from '../config';
+
+const MODAL_DISPLAY_INTERVAL = 2 * 24 * 60 * 60 * 1000; // 2 days in milliseconds
+
+interface HackathonModalProps {
+  onRequestClose: () => void;
+}
+
+const HackathonModal: React.FC<HackathonModalProps> = ({ onRequestClose }) => {
+  const [isOpen, setIsOpen] = useState<boolean>(false);
+  const [assetsLoaded, setAssetsLoaded] = useState<boolean>(false);
+
+  useEffect(() => {
+    const fontFamilies = ['Orbitron', 'Poppins', 'Motorblock'];
+
+    const imageUrls = [
+      '/Hackathon/little-guy.png',
+      '/Hackathon/top-left.svg',
+      '/Hackathon/top-right.svg',
+      '/Hackathon/bottom-left.svg',
+    ];
+
+    const loadFonts = async () => {
+      try {
+        await Promise.all(
+          fontFamilies.map((family) => document.fonts.load(`1em "${family}"`))
+        );
+      } catch (error) {
+        console.error('Failed to load fonts:', error);
+      }
+    };
+
+    const loadImages = async () => {
+      try {
+        await Promise.all(
+          imageUrls.map((url) => {
+            return new Promise<void>((resolve, reject) => {
+              const img = new window.Image();
+              img.onload = () => resolve();
+              img.onerror = () =>
+                reject(new Error(`Failed to load image: ${url}`));
+              img.src = url;
+            });
+          })
+        );
+      } catch (error) {
+        console.error('Failed to load images:', error);
+      }
+    };
+
+    Promise.all([loadFonts(), loadImages()])
+      .then(() => setAssetsLoaded(true))
+      .catch((error) => console.error('Failed to load assets:', error));
+  }, []);
+
+  useEffect(() => {
```

Comment:
So assets loaded can only change once why are these two seperate `useEffects` this seems like a state management issue. seems like it would be better to have this just follow in the regular useEffect would it not be?